src/mlpack/core/data/scaler_methods/whitening.hpp

```diff
@@ -0,0 +1,147 @@
+/**
+ * @file whitening.hpp
+ * @author Jeffin Sam
+ *
+ * Whitening scaling to scale features, Using PCA and ZCA Whitening.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_WHITENING_SCALE_HPP
+#define MLPACK_CORE_DATA_WHITENING_SCALE_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/lin_alg.hpp>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * A simple Whitening class.
+ *
+ * Whitens a matrix using the eigendecomposition of the covariance matrix.
+ * Whitening means the covariance matrix of the result is the identity matrix.
+ * 
+ * For whitening related formula and more info, check the link below.
+ * http://ufldl.stanford.edu/tutorial/unsupervised/PCAWhitening/
+ *
+ * @code
+ * arma::mat input;
+ * Load("train.csv", input);
+ * arma::mat output;
+ *
+ * // Scale the features using PCA.
+ * Whitening scale;
+ * scale.PCA(input, output);
+ *
+ * // Scale the features using ZCA.
+ * scale.ZCA(input, ouput);
```

Comment:
Design-wise, this is different than the other classes because it doesn't have a `Transform()` method.  I'd suggest splitting this into two classes (`PCAWhitening` and `ZCAWhitening`), and providing a `Transform()` method just like the other classes do.

Reply:
Made appropriate changes, and adapted the same api as others.